src/main/java/org/rulez/demokracia/pdengine/WinnerCalculator.java

```diff
@@ -0,0 +1,12 @@
+package org.rulez.demokracia.pdengine;
+
+import java.util.List;
+
+public class WinnerCalculator {
+
+
+	public List<String> calculateWinners(final BeatTable beatTable) {
```

Comment:
I am not entirely sure that the trait pattern as it is defined right now is the best way to go.
However according to that this should be a default method of the interface, and you would spare a dependency injection in the main class - which opens possibilities for errors. If you can argue for this solution, do that, otherwise let's stick to the rules what we have now.

Reply:
You are right. I violated the Dependency Inversion Principle. We should create an interface, but I'm not sure about default methods are the best practice, since according to DIP:
- All member variables in a class must be interfaces or abstracts.
- All concrete class packages must connect only through interface or abstract class packages.
- No class should derive from a concrete class.
- **No method should override an implemented method.**
- All variable instantiation requires the implementation of a creational pattern such as the factory method or the factory pattern, or the use of a dependency-injection framework.